cache/demux.go

```diff
@@ -167,16 +177,32 @@ func (d *demux) updateStoreLocked(events []*clientv3.Event) {
 	d.maxRev = events[len(events)-1].Kv.ModRevision
 }
 
-func (d *demux) broadcastLocked(events []*clientv3.Event) {
-	firstRev := events[0].Kv.ModRevision
-	lastRev := events[len(events)-1].Kv.ModRevision
+func (d *demux) broadcastLocked(resp clientv3.WatchResponse) {
+	if resp.IsProgressNotify() {
+		for w, nextRev := range d.activeWatchers {
+			if nextRev >= resp.Header.Revision {
+				continue
+			}
+			if !w.enqueueResponse(clientv3.WatchResponse{
+				Header: etcdserverpb.ResponseHeader{
+					Revision: resp.Header.Revision,
+				},
+			}) { // overflow → lagging
+				d.laggingWatchers[w] = nextRev
```

Comment:
Maybe let's rethink it. Is missing a progress notification reason to kick watcher to lagging?  Progress notification are optional not guaranteed signals about progress. If lost they don't require replay. I don't think there is a reason to drop watcher to lagging.